pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -0,0 +1,270 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, List, Optional, Sequence, Union
+
+from deprecate.utils import void
+from torch.utils.data.dataloader import DataLoader
+
+import pytorch_lightning as pl
+from pytorch_lightning.loops.dataloader.dataloader_loop import DataLoaderLoop
+from pytorch_lightning.loops.evaluation_epoch_loop import EvaluationEpochLoop
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.types import EPOCH_OUTPUT
+
+
+class EvaluationDataLoaderLoop(DataLoaderLoop):
+    """Loops over all dataloaders for evaluation."""
+
+    def __init__(self):
+        super().__init__()
+        self._max_batches: Optional[Union[int, Sequence[int]]] = None
+        self.outputs = []
+        self.evaluation_loop = EvaluationEpochLoop()
+
+        self._val_results = ResultCollection(training=False)
+        self._test_results = ResultCollection(training=False)
+
+    @property
+    def num_dataloaders(self) -> int:
+        """Returns the total number of dataloaders"""
+        return self._get_num_dataloaders(self.dataloaders)
```

Comment:
Should we remove this function call? Is it used anywhere else?

Reply:
good call, integrated here in the property now!